src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -794,6 +813,87 @@ namespace
             {
             }
 
+            virtual bool accept_scattering(
```

Comment:
We shouldn't need any virtual methods since the visitors are passed as template parameters (precisely to avoid virtual methods, actually).